CONTRIBUTING.md

```diff
@@ -98,11 +96,45 @@ or more information on the GitHub Actions and Pull Request Workflow, please see
 Linting, styling, and cleaning checks are automatically performed on pull requests using GitHub Actions. This ensures that contributed code meets standard Python coding standards before it's merged into the main branch.
 
 1. **Pull Request Process**: When you open a pull request, GitHub Actions will automatically trigger linting, styling, and cleaning checks on the changes made within the `benchmarking` directory.
-
 2. **Approval Requirement**: In order to merge a pull request, it must pass the GitHub Actions workflow test. This ensures that all contributions adhere to our coding standards and maintain consistency throughout our `benchmarking` repository.
-
 3. **Interpreting Results**: If linting fails on your pull request, review the output to identify and fix any issues. You'll need to address these issues before the pull request can be approved and merged. In case of repeated failures or failures within the GitHub Actions workflow files, please reach out to one of the repository maintainers from the [Maintainers.md](MAINTAINERS.md).
 
 #### Automated Commit by GitHub Actions
 
-The GitHub Actions workflow also automatically makes a commit with the message ```*** AUTOMATED COMMIT | Applied Code Formatting and Cleanup ✨ 🍰 ✨***``` authored by ```[anirudTT]``` when it performs code formatting and cleanup. If you open a pull request and subsequently push more changes, we suggest **rebasing or pulling again** from the pull request branch before pushing your changes again to avoid conflicts.
+The GitHub Actions workflow also automatically makes a commit with the message `*** AUTOMATED COMMIT | Applied Code Formatting and Cleanup ✨ 🍰 ✨***` authored by `[anirudTT]` when it performs code formatting and cleanup. If you open a pull request and subsequently push more changes, we suggest **rebasing or pulling again** from the pull request branch before pushing your changes again to avoid conflicts.
+
+#### SPDX License Compliance Check
+
+Our project enforces SPDX license compliance for all Python files through automated checks integrated within our GitHub Actions workflows. These workflows, `check-license-year.yml` and `license-checker.yml`, automatically verify the presence and accuracy of SPDX license headers and the specified license year in Python files included in pull requests.
+
+- **License Year Validation**: The `check-license-year.yml` workflow checks for the correct SPDX license year (`2024`) in newly changed Python files. If files lacking the correct year or license identifier are detected, the workflow comments on the PR with details on the missing elements and fails to highlight the need for compliance.
```

Comment:
Anyway for us to automate the year it verifies against? Somehow extract the current year from a system time.

This way we don't have to worry about updating the repo every new year.